packages/react-aria-components/src/utils.tsx

```diff
@@ -132,11 +134,11 @@ export function useRenderProps<T>(props: RenderPropsHookOptions<T>) {
 
     return {
       className: computedClassName ?? defaultClassName,
-      style: computedStyle,
-      children: computedChildren,
+      style: (computedStyle || defaultStyle) ? {...defaultStyle, ...computedStyle} : undefined,
```

Comment:
note that defaultStyle works slightly differently to defaultClassName and defaultChildren: it is _merged_ with the computed style rather than being overridden. I think this makes sense otherwise whenever anyone used a `style` render prop they would need to remember to merge the defaults manually. Since the merge is shallow, it is still possible to override/remove a style property from the defaults by setting it to undefined.